base/twiceprecision.jl

```diff
@@ -456,6 +456,9 @@ step(r::StepRangeLen{T,TwicePrecision{T},TwicePrecision{T}}) where {T} = T(r.ste
 range_start_step_length(a, st::IEEEFloat, len::Integer) =
     range_start_step_length(oftype(st, a), st, len)
 
+range_start_step_length(a::IEEEFloat, st, len::Integer) =
```

Comment:
```suggestion
range_start_step_length(a::IEEEFloat, st::Real, len::Integer) =
```

Should we try to limit this to Reals? Looking back at the old ranges.jl code, that seemed to be the intent. I think we don't want to end up trying to catch this case, for example:
```julia
julia> range(0.0, step=1im, length=10) |> typeof
StepRangeLen{ComplexF64, Float64, Complex{Int64}, Int64}
```

Reply:
Makes sense, thanks!